src/assets/scripts/client/airline/AirlineController.js

```diff
@@ -33,35 +49,80 @@ export default class AirlineController {
+     * for faster processing of aircraft commands by using only a `flightNumber` for a command.
+     *
      * @for AirlineController
-     * @method airline_get
-     * @param icao {string}
-     * return {AirlineModel|null}
+     * @method generateFlightNumberWithAirlineModel
+     * @param airlineModel {AirlineModel}
+     * @return flightNumber {string}
      */
-    airline_get(icao) {
-        icao = icao.toLowerCase();
+    generateFlightNumberWithAirlineModel(airlineModel) {
+        if (!(airlineModel instanceof AirlineModel)) {
+            throw new TypeError('Invalid parameter. Expected airlineModel to be an instance of AirlineModel');
+        }
 
-        if (!_has(this.airline.airlines, icao)) {
-            this.addAirline(icao);
+        const flightNumber = airlineModel.generateFlightNumber();
+
+        if (this._hasFlightNumber(flightNumber)) {
+            this.generateFlightNumberWithAirlineModel(airlineModel);
         }
 
-        return this.airline.airlines[icao];
+        airlineModel.addFlightNumberToInUse(flightNumber);
+
+        return flightNumber;
     }
 
     /**
-     * @for airlineController
-     * @method addAirline
-     * @param icao {string}
+     * Remove a provided flightNumber from an airline's list of `flightNumbersInUse`
+     *
+     * Used when an aircraft leaves controlled airspace. Removing a flightNumber
+     * from the list allows it to be used again by another aircraft
+     *
+     * @for AirlineCollection
+     * @method removeFlightNumberFromList
+     * @param airline {string}
+     * @param callsign {string}
+     */
+    removeFlightNumberFromList(airline, callsign) {
+        const airlineModel = this.findAirlineById(airline);
+
+        if (_isNil(airlineModel)) {
+            // TODO: should we throw here or console.error?
+            return;
+        }
+
+        airlineModel.removeFlightNumber(callsign);
+    }
+
+    /**
+     * Reset session specific class properties
+     *
+     * Used when changing airports. Clears flightNumbersInUse and
+     * any other properties specific to a current session at a
+     * specific airport
+     *
+     * @for AirlineController
+     * @method reset
+     */
+    reset() {
+        this.airlineCollection.reset();
+    }
+
+    /**
+     * Used to determine if a provided `flightNumber` is already in use
+     *
+     * @for AirlineController
+     * @method _hasFlightNumber
+     * @param flightNumber {string}
+     * @returns {boolean}
      */
-    addAirline(icao) {
-        console.error('.addAirline() is a deprecated method');
-        const airlineToAdd = new Airline(
-            icao,
-            {
-                url: `assets/airlines/${icao}.json`
-            }
-        );
+    _hasFlightNumber(flightNumber) {
+        const invalidIndex = -1;
 
-        this.airline.airlines[icao] = airlineToAdd;
+        return this.flightNumbers.indexOf(flightNumber) !== invalidIndex;
```

Comment:
It is, I've used it somewhere in one of my previous PRs. But that's cool beans, just throwing it out there